### PR TITLE
[Enhancement] Force enable replicated storage when ingestion need shuffle

### LINKED ISSUE #24419: [Bug] insert into select limit result not stable
### Steps to reproduce the behavior (Required)
1. Create table `(replicated_storage=false)` and insert one row into this table:
```
create database test;
use test;
CREATE TABLE `test_table1` (
  `id` int(11) NOT NULL COMMENT "",
  `call_id` varchar(150) NULL COMMENT "",
  `keyword` varchar(65000) NULL COMMENT "",
  `content` varchar(65000) NULL COMMENT "",
  `attribute` varchar(60) NULL COMMENT "",
  `c_time` varchar(60) NULL COMMENT "",
  `match_title` varchar(300) NULL COMMENT "",
  `node_type` tinyint(4) NULL COMMENT "",
  `node_id` int(11) NULL COMMENT "",
  `correction` tinyint(4) NULL COMMENT "",
  `bak_1` varchar(30) NULL COMMENT "",
  `intention_info` varchar(765) NULL COMMENT "",
  INDEX idx_call_id (`call_id`) USING BITMAP COMMENT 'idx_call_id'
) ENGINE=OLAP
PRIMARY KEY(`id`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`id`) BUCKETS 10
PROPERTIES (
"replication_num" = "3",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"compression" = "LZ4"
);

insert into test_table1 select MURMUR_HASH3_32(rand()), substr(md5(rand()),-30), substr(md5(rand()),-30), substr(md5(rand()),-30),substr(md5(rand()),-30),1685808000,substr(md5(rand()),-30), 1,  MURMUR_HASH3_32(rand()),1,substr(md5(rand()),-30),substr(md5(rand()),-30);
```

2. execute insert into select many times:
```
use test;
insert into test.test_table1  (
  `id`,
  `call_id` ,
  `keyword`,
  `content` ,
  `attribute` ,
  `c_time` ,
  `match_title` ,
  `node_type` ,
  `node_id`,
  `correction`,
  `bak_1` ,
  `intention_info`
  ) select   `id`  + MURMUR_HASH3_32(rand()) % 101 , `call_id` ,
  `keyword`,
  `content` ,
  `attribute` ,
  cast(`c_time` as int)+ 20000 ,
  `match_title` ,
  `node_type` ,
  `node_id`,
  `correction`,
  `bak_1` ,
  `intention_info` from test.test_table1 limit 1000000;
```
### Expected behavior (Required)
Result is stable.
```
mysql> select count(1) from test.test_table1 where c_time = 1686268000;
+----------+
| count(1) |
+----------+
|      364 |
+----------+
1 row in set (0.01 sec)

mysql> select count(1) from test.test_table1 where c_time = 1686268000;
+----------+
| count(1) |
+----------+
|      364 |
+----------+
1 row in set (0.01 sec)
```

### Real behavior (Required)
Result not stable:
```
mysql> select count(1) from test.test_table1 where c_time = 1686268000;
+----------+
| count(1) |
+----------+
|      364 |
+----------+
1 row in set (0.01 sec)

mysql> select count(1) from test.test_table1 where c_time = 1686268000;
+----------+
| count(1) |
+----------+
|      367 |
+----------+
1 row in set (0.01 sec)
```
This result isn't consistent between different replica:
```
mysql> select * from test.test_table1 where id = -1899502016\G
*************************** 1. row ***************************
            id: -1899502016
       call_id: 88665341ded21bf0f8be0f4a0435ee
       keyword: 278be057211a0a62553f524a13105c
       content: b588727a57b252bbd9aea4b49b2731
     attribute: 507f6d5d11f9cf77fb8f1cab99a000
        c_time: 1686348000
   match_title: 1fe5c45def15589751dba027e64335
     node_type: 1
       node_id: -1135081274
    correction: 1
         bak_1: 99936d89c1cd2a72dd467507b27859
intention_info: d0ed02ba34e98987c03afce7cc96ea
1 row in set (0.00 sec)

mysql> select * from test.test_table1 where id = -1899502016\G
*************************** 1. row ***************************
            id: -1899502016
       call_id: 88665341ded21bf0f8be0f4a0435ee
       keyword: 278be057211a0a62553f524a13105c
       content: b588727a57b252bbd9aea4b49b2731
     attribute: 507f6d5d11f9cf77fb8f1cab99a000
        c_time: 1686268000
   match_title: 1fe5c45def15589751dba027e64335
     node_type: 1
       node_id: -1135081274
    correction: 1
         bak_1: 99936d89c1cd2a72dd467507b27859
intention_info: d0ed02ba34e98987c03afce7cc96ea
1 row in set (0.01 sec)
```
### StarRocks version (Required)
 - 2.5.6

### PR DESCRIPTION
**Pull request by meegoo**
Fixes #24419

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 35 / 35 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/load/loadv2/LoadingTaskPlanner.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/LoadPlanner.java|22|22|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|7|7|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -1940,6 +1940,12 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true)
     public static boolean enable_shuffle_load = true;
 
+    /**
+     * Eliminate shuffle load by replicated storage
+     */
+    @ConfField(mutable = true)
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1946]: Why to need the configuration?

Comment (meegoo) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1946]: replicated_storage is forced to open here, if there is a bug, we have to find a way to turn it off.

Comment (chaoyli) [reply fe/fe-core/src/main/java/com/starrocks/common/Config.java:1946]: If you are concerned about the `replicated_storage`, why not turn the `replicated_storage` off when there is a bug?

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**